sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -253,46 +253,10 @@ public virtual Response<ConfigurationSetting> SetConfigurationSetting(string key
         public virtual async Task<Response<ConfigurationSetting>> SetConfigurationSettingAsync(ConfigurationSetting setting, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default)
         {
-            if (setting == null)
-                throw new ArgumentNullException($"{nameof(setting)}");
-
-            MatchConditions requestOptions = default;
-            if (onlyIfUnchanged)
-            {
-                requestOptions = new MatchConditions();
-                requestOptions.IfMatch = setting.ETag;
-            }
-
-            return await SetConfigurationSettingAsync(setting, requestOptions, cancellationToken).ConfigureAwait(false);
-        }
-
-        /// <summary>
-        /// Creates a <see cref="ConfigurationSetting"/> if it doesn't exist or overwrites the existing setting in the configuration store.
-        /// </summary>
-        /// <param name="setting">The <see cref="ConfigurationSetting"/> to create.</param>
-        /// <param name="onlyIfUnchanged">If set to true and the configuration setting exists in the configuration store, overwrite the setting
-        /// if the passed-in <see cref="ConfigurationSetting"/> is the same version as the one in the configuration store.  The setting versions
-        /// are the same if their ETag fields match.  If the two settings are different versions, this method will throw an exception to indicate
-        /// that the setting in the configuration store was modified since it was last obtained by the client.</param>
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>A response containing the <see cref="ConfigurationSetting"/> written to the configuration store.</returns>
-        public virtual Response<ConfigurationSetting> SetConfigurationSetting(ConfigurationSetting setting, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default)
-        {
-            if (setting == null)
-                throw new ArgumentNullException($"{nameof(setting)}");
-
-            MatchConditions requestOptions = default;
-            if (onlyIfUnchanged)
-            {
-                requestOptions = new MatchConditions();
-                requestOptions.IfMatch = setting.ETag;
-            }
+            Argument.AssertNotNull(setting, nameof(setting));
 
-            return SetConfigurationSetting(setting, requestOptions, cancellationToken);
-        }
+            MatchConditions requestOptions = onlyIfUnchanged ? new MatchConditions { IfMatch = setting.ETag } : default;
 
-        internal virtual async Task<Response<ConfigurationSetting>> SetConfigurationSettingAsync(ConfigurationSetting setting, MatchConditions requestOptions, CancellationToken cancellationToken = default)
-        {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.Data.AppConfiguration.ConfigurationClient.SetConfigurationSetting");
```

Comment:
DiagnosticScope should go before MatchConditions is created.

Reply:
`MatchConditions` is actually needed only inside `CreateSetRequest`. Fixed.